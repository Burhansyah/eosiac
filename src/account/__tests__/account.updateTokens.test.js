const cloneDeep = require(`lodash/cloneDeep`)
const Account = require(`../index`)
const setup = require(`./setup`)

const {env, accountConfig, existingAccount} = setup

describe(`account.updateTokens`, () => {
    it(`throws when account does not exist yet`, async () => {
        expect.assertions(1)
        const a = new Account(`test`, accountConfig)
        a.currentState = {}
        expect(a.updateTokens({env})).rejects.toThrow(/does not exist/i)
    })

    it(`returns no actions to run when exact same token`, async () => {
        expect.assertions(1)
        const a = new Account(`test`, accountConfig)
        a.currentState = cloneDeep(existingAccount)
        a.currentState.tokens = [`100.0000 EOS`]
        const actions = await a.updateTokens({env})
        expect(actions).toBeFalsy()
    })

    it(`returns positive transfer action`, async () => {
        expect.assertions(1)
        const a = new Account(`test`, accountConfig)
        a.currentState = cloneDeep(existingAccount)
        const actions = await a.updateTokens({env})
        expect(actions).toMatchSnapshot()
    })

    it(`returns negative transfer action`, async () => {
        expect.assertions(1)
        const a = new Account(`test`, accountConfig)
        a.currentState = cloneDeep(existingAccount)
        a.currentState.tokens = [`200.0000 EOS`]
        const actions = await a.updateTokens({env})
        expect(actions).toMatchSnapshot()
    })
})
