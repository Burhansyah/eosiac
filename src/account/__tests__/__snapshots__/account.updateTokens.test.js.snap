// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account.updateTokens returns negative transfer action 1`] = `
Array [
  Object {
    "account": "eosio.token",
    "authorization": Array [
      Object {
        "actor": "test",
        "permission": "active",
      },
    ],
    "data": Object {
      "from": "test",
      "memo": "Fund token",
      "quantity": "100.0000 EOS",
      "to": "fundsmngr",
    },
    "name": "transfer",
  },
]
`;

exports[`account.updateTokens returns positive transfer action 1`] = `
Array [
  Object {
    "account": "eosio.token",
    "authorization": Array [
      Object {
        "actor": "fundsmngr",
        "permission": "active",
      },
    ],
    "data": Object {
      "from": "fundsmngr",
      "memo": "Fund token",
      "quantity": "90.0000 EOS",
      "to": "test",
    },
    "name": "transfer",
  },
]
`;
